concent_api/core/views.py

```diff
@@ -1447,6 +1448,135 @@ def store_subtask(
+):
+
+    
+    clients_subtask_list = Subtask.objects.filter(
+        delivered = False,
+        state__in = Subtask.SubtaskState.SUBTASK_ACTIVE_STATES,
+        Q(requestor__public_key = client_public_key) | Q(provider__public_key = client_public_key)
+    )
+
+    if not clients_subtask_list.exist():
+        logging.log_no_changes_in_subtask_states(client_public_key)
+    else:
+        if clients_subtask_list.count() != 0:
+            update_states_of_subtasks(clients_subtask_list)
+            logging.log_changes_in_subtask_states(
+                client_public_key,
+                clients_subtask_list.count(),
+            )
+
+
+def update_states_of_subtasks(
+    subtask_list:       list,
+    client_public_key:  str,
+):
+    current_time = get_current_utc_timestamp()
+
+    for subtask in subtask_list:
+        if subtask.next_deadline.timestamp() <= current_time:
+            if subtask.state == Subtask.SubtaskState.FORCING_REPORT:
+                subtask = change_subtask_state(
+                    subtask = subtask,
+                    state   = Subtask.SubtaskState.REPORTED
+                )
+                store_or_update_receive_queue(
+                    response_type   = MessageInbox.ResponseType.VerdictReportComputedTask,
+                    client          = client_public_key,
+                    queue           = MessageInbox.MessageInboxType.ReceiveOutOfBand,
+                    delivered       = False,
+                    subtask         = subtask,
+                )
+            elif subtask.state == Subtask.SubtaskState.FORCING_RESULT_TRANSFER:
+                subtask = change_subtask_state(
+                    subtask = subtask,
+                    state   = Subtask.SubtaskState.FAILED
+                )
+                store_or_update_receive_queue(
+                    response_type   = MessageInbox.ResponseType.ForceGetTaskResultRejected,
+                    client          = client_public_key,
+                    queue           = MessageInbox.MessageInboxType.Receive,
+                    delivered       = False,
+                    subtask         = subtask,
+                )
+            elif subtask.state == Subtask.SubtaskState.FORCING_ACCEPTANCE:
+                subtask = change_subtask_state(
+                    subtask = subtask,
+                    state   = Subtask.SubtaskState.REJECTED
+                )
+                store_or_update_receive_queue(
+                    response_type   = MessageInbox.ResponseType.ForceSubtaskResultsResponse,
+                    client          = client_public_key,
+                    queue           = MessageInbox.MessageInboxType.Receive,
+                    delivered       = False,
+                    subtask         = subtask,
+                )
+            # TODO when last use case will be implemented
+            # elif subtask.state == Subtask.SubtaskState.ADDITIONAL_VERIFICATION:
+
+
+def change_subtask_state(
```

Comment:
Please check if you can use `update_subtask` function instead of this, it should have functionality which can do this too.